diff -Nru zabbix-1.8.11.orig/frontends/php/include/config.inc.php zabbix-1.8.11/frontends/php/include/config.inc.php
--- zabbix-1.8.11.orig/frontends/php/include/config.inc.php	2012-03-20 11:15:17.000000000 +0200
+++ zabbix-1.8.11/frontends/php/include/config.inc.php	2012-04-03 15:55:34.000000000 +0300
@@ -80,7 +80,7 @@
 	require_once('include/html.inc.php');
 	require_once('include/copt.lib.php');
 	require_once('include/profiles.inc.php');
-	require_once('conf/maintenance.inc.php');
+	require_once('/etc/zabbix/web/maintenance.inc.php');
 
 	require_once('include/nodes.inc.php');
 	require_once('include/hosts.inc.php');
@@ -119,8 +119,7 @@
 	$ZBX_LOCALNODEID = 0;
 	$ZBX_LOCMASTERID = 0;
 
-	$ZBX_CONFIGURATION_FILE = './conf/zabbix.conf.php';
-	$ZBX_CONFIGURATION_FILE = realpath(dirname($ZBX_CONFIGURATION_FILE)).'/'.basename($ZBX_CONFIGURATION_FILE);
+	$ZBX_CONFIGURATION_FILE = '/etc/zabbix/web/zabbix.conf.php';
 
 // Include Tactical Overview modules
 	require_once('include/locales.inc.php');
